### ISSUE TITLE
Add broader concept as qualifier to terms alphabetical index

### ISSUE THREAD

**Issue by osma** [NatLibFi/Skosmos#587]
## At which URL did you encounter the problem?
http://dev.finto.fi/iptc/fi/index?clang=en

## What steps will reproduce the problem?
1. Go to IPTC
2. Look at the alphabetical index

## What is the expected output? What do you see instead?

The problem is that there are identical terms, e.g. "1000 m" is listed three times. We could automatically add a qualifier such as "(speed skating)" to these. It should be controlled by a configuration option.

**Comment by kinow** [comment]
Hmmm, this one looked easy to fix, but spending 20 minutes reading it and reproducing locally, I am not so sure which is the best way to tackle it. 

The data returned for the alphabetical sidebar contains only the label and URI I think. We would have to add a configuration parameter used in the UI and also for running the query?

If that parameter is true (and I assume that parameter would be a per vocabulary setting), then the query would bring also the array of broader terms.

And in the UI, if that parameter is true, we would display the concept followed by (broader1, broader2, ...)?

Would it be applied to all concepts, or only to repeated ones? If the latter, we also would have - probably in the controller I guess - a filter to change the `prefLabel` maybe, adding the broader concepts... as the controller is dealing with a limited list or in-order concepts, that could work?

Cheers
Bruno

**Comment by osma** [comment]
> We would have to add a configuration parameter used in the UI and also for running the query? If that parameter is true (and I assume that parameter would be a per vocabulary setting), then the query would bring also the array of broader terms.

Yes, this would be a per vocabulary setting. It would affect the SPARQL query, to make it include the broader concepts (or at least their labels) too.

> And in the UI, if that parameter is true, we would display the concept followed by (broader1, broader2, ...)?

Right.

> Would it be applied to all concepts, or only to repeated ones? If the latter, we also would have - probably in the controller I guess - a filter to change the prefLabel maybe, adding the broader concepts... as the controller is dealing with a limited list or in-order concepts, that could work?

I think it would be better for users to apply this only to concepts which look like duplicates in the current alphabetical index because they have the same label but are actually different. So doing it with a list of in-order concepts, looking only within a small window, could work.

In the rendered listing, I think it would be best not to add the broader concept qualifier not to the prefLabel itself (which is a link to the concept page) but keep it separate. So in the example given in the OP, the final list could look like this:

  * [100 yards](http://dev.finto.fi/iptc/en/page/15005054)
  * [1000 m](http://dev.finto.fi/iptc/en/page/15005005) (athletics, track and field)
  * [1000 m](http://dev.finto.fi/iptc/en/page/15015004) (canoeing and kayaking)
  * [1000 m](http://dev.finto.fi/iptc/en/page/15056002) (speed skating)
  * [1000 yards](http://dev.finto.fi/iptc/en/page/15005064)

**Comment by kouralex** [comment]
In some other vocabularies, e.g., [Metatietosanasto](http://finto.fi/mts/fi/index), this distinction should be done (qualifier should be selected) based on skos:member relationship. See http://finto.fi/mts/en/page/m2302 and http://finto.fi/mts/en/page/m238.

 Other such relationships may exist as well and should be taken into consideration.
